components/kyma-environment-broker/internal/process/deprovisioning/initialisation.go

```diff
@@ -114,6 +114,10 @@ func (s *InitialisationStep) run(operation internal.DeprovisioningOperation, log
 	instance, err := s.instanceStorage.GetByID(operation.InstanceID)
 	switch {
 	case err == nil:
+		// If the instance is currently suspended, let this deprovision operation clean up the instance
+		if instance.Parameters.ErsContext.Active != nil && !*instance.Parameters.ErsContext.Active {
```

Comment:
in general it should work. I'm afraid if the suspension failed (something was not removed) - such resource will be orphaned.

Reply:
As far as I can see on just need to add the `&& instance.RuntimeID == ""` to the condition, because if it's empty, it means that the runtime was either successfully removed before during suspension, or never existed (due to some failure). If the `RuntimeID` is not empty, the operation could proceed until `Remove_Runtime`, triggering new `deleteRuntime` in provisioner and polling the status. 

What do you think?